File: java-web-spring-boot-3/src/main/java/com/vndevteam/javawebspringboot3/util/DateTimeUtils.java
package com.vndevteam.javawebspringboot3.util;

import java.time.LocalDateTime;
import java.time.ZoneId;
import java.time.ZonedDateTime;
import java.time.format.DateTimeFormatter;
import java.util.TimeZone;

public class DateTimeUtils {
    public static ZonedDateTime getNow() {
        return ZonedDateTime.now(TimeZone.getDefault().toZoneId());
    }

    public static String getNow(String format) {
        DateTimeFormatter df = DateTimeFormatter.ofPattern(format);
        return df.format(getNow());
    }

    public static LocalDateTime toZone(LocalDateTime localDateTime, ZoneId zoneId) {
        return localDateTime.atZone(zoneId).toLocalDateTime();
    }
}

File: kotlin-web-spring-boot-3/src/main/kotlin/com/vndevteam/kotlinwebspringboot3/util/DateTimeUtils.kt
package com.vndevteam.kotlinwebspringboot3.util

import java.time.LocalDateTime
import java.time.ZoneId
import java.time.ZonedDateTime
import java.time.format.DateTimeFormatter
import java.util.*

class DateTimeUtils {
    companion object {
        fun getNow(): ZonedDateTime {
            return ZonedDateTime.now(TimeZone.getDefault().toZoneId())
        }

        fun getSystemNow(format: String): String {
            val df = dateTimeFormatter(format)
            return df.format(LocalDateTime.now())
        }

        fun toZone(source: LocalDateTime): LocalDateTime {
            return LocalDateTime.from(source)
                .atZone(ZoneId.systemDefault()) // current zone
                .withZoneSameInstant(ZoneId.of("Asia/Tokyo")) // new zone
                .toLocalDateTime()
        }

        fun dateTimeFormatter(format: String): DateTimeFormatter {
            return DateTimeFormatter.ofPattern(format, Locale.JAPAN)
        }
    }
}
